### PR TITLE
Fix mpas_atm compute_full_pressure

### LINKED ISSUE #251: bug: Constants in the model_mod for MPAS
I noticed that an atmospheric constant (specifically Cp or Specific heat of dry air at constant pressure) in model_mod.f90 in the mpas_atm directory does not match the value found in the MPAS source code.

DART Cp Constant Value (line 215 in model_mod.f90): 1003.0
MPAS Cp Constant Value (MPAS-Release/src/framework/mpas_constants.F: line 33): 1004.5

I believe you would want these constant to match up when converting potential temperature to temperature in the forward operator function (theta_to_tk line 7428).

### PR DESCRIPTION
**Pull request by syha**
## Description:
The HofX (e.g., observation operators) described in model_mod.f90 computes the pressure field based on the equation of state to find the vertical locations of observations at pressure levels (ex. radiosondes).  Because the "rho" used in model_mod indicates dry air density (e.g., rho_d), not air density (= rho_d + rho_v), we need to add the second order term for water vapor mixing ratio (qv) to compute the full pressure accounting for the moisture effect correctly. 
Also, a few parameters are updated based on MPAS/src/framework/mpas_constants.F.

### Fixes issue
     pressure = rho * rgas * tk * (1.0_r8 + 1.61_r8 * qv_nonzero)
=> pressure = rho * rgas * tk * (1.0_r8 + rvord * qv_nonzero + rvordm1 * qv_nonzero**2)

fixes #251

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Documentation changes needed?

- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.

### Tests
Not tested with data yet, but compiled successfully (on cheyenne).

## Checklist for merging

- [ ] Updated changelog entry
- [ ] Documentation updated
- [ ] Update conf.py

## Checklist for release
- [ ] Merge into main
- [ ] Create release from the main branch with appropriate tag
- [ ] Delete feature-branch

### CODE REVIEW

**Comment by hkershaw-brown** [comment]
Thanks for this Soyoung. 

For the change in constants cv and cp does this apply to mpas versions 5.0 onwards?

**Comment by syha** [comment]
> Thanks for this Soyoung.
> 
> For the change in constants cv and cp does this apply to mpas versions 5.0 onwards?

It looks like MPAS Versions 5 through 7 have these new parameters.

**Comment by hkershaw-brown** [comment]
great thanks!

**Comment by hkershaw-brown** [comment]
waiting for science review on this.

File: models/mpas_atm/model_mod.f90 [hunk 1]
```diff
@@ -209,14 +209,15 @@ module model_mod
...
 ! Real (physical) constants as defined exactly in MPAS.
-! redefined here for consistency with the model.
-real(r8), parameter :: rgas = 287.0_r8
-real(r8), parameter :: rv = 461.6_r8
-real(r8), parameter :: cp = 1003.0_r8
-real(r8), parameter :: cv = 716.0_r8
+! redefined here for consistency with the model (MPAS/src/framework/mpas_constants.F).
+real(r8), parameter :: rgas = 287.0_r8  ! = R_d (Gas constant for dry air [J kg-1 K-1])
+real(r8), parameter :: rv = 461.6_r8    ! = R_v (Gas constant for water varpor [J kg-1 K-1])
+real(r8), parameter :: cp = 7.*rgas/2.  ! = 1004.5
```
Comment (kdraeder) [inline models/mpas_atm/model_mod.f90:215]: Another comment might be helpful here: "For an ideal diatomic gas, cp:cv:R is 7:5:2"

**Review by kdraeder — COMMENTED** [review]
I have one suggestion and one major sticking point.

**Comment by nancycollins** [comment]
just a note to say that i talked with soyoung about this today.  she's sure the equation is correct; mpas uses mixing ratio and it is just the comment about the equation that needs to be clarified.  at some point soon she is going to merge in the current main branch and update the code for this pull request.

**Comment by kdraeder** [comment]
@nancycollins and @syha 
which equation are you saying is correct?  
1) The ideal gas law that's in the _new_ comment (which looks correct to me), or 
2) the equation that's in the updated code:
pressure = rho * rgas * tk * (1.0_r8 + rvord * qv_nonzero + rvordm1 * qv_nonzero**2)?
I can't derive 2) from 1).  1) has moisture in it, and when it's rewritten with the available fortran variables,
it looks like the old code.
Does qv_nonzero have a new definition which I'm not seeing, or was it used incorrectly before?
There seems to be only one file changed, and qv_nonzero was not one of the changes.

**Comment by syha** [comment]
@kdraeder, thanks for your examination with patience! 
You are right. We should go with the original equation (1) for the full pressure. 
Let me do a new PR for the constant updates only.

**Comment by nancycollins** [comment]
in the interest of putting this issue to bed - soyoung says that she was looking at how jedi handles computing pressure for mpas.   they do a transform which requires a different equation (which includes the squared term).  she says the original equation and comment are correct for the dart model_mod, and that is why this pull request is closed.  the original code and comment should remain unchanged.   the constants need to be updated and that's happening in a separate pull request (#521).

**Pull request closed without merging**